src/wallet/rpcwallet.cpp

```diff
@@ -4267,7 +4282,57 @@ UniValue z_viewtransaction(const UniValue& params, bool fHelp)
         addMemo(entry, memo);
         outputs.push_back(entry);
     }
-    // TODO unified addresses, orchard, see #5186
+
+    std::vector<uint256> ovksVector(ovks.begin(), ovks.end());
+    OrchardActions orchardActions = pwalletMain->GetOrchardWallet().GetTxActions(wtx, ovksVector);
+
+    // Orchard spends
+    for (auto & pair  : orchardActions.GetSpends()) {
+        // TODO test with Orchard spends after rebased on feature/wallet-orchard
+        auto actionIdx = pair.first;
+        OrchardActionSpend orchardActionSpend = pair.second;
+        auto outpoint = orchardActionSpend.GetOutPoint();
+        auto receivedAt = orchardActionSpend.GetReceivedAt();
+        auto noteValue = orchardActionSpend.GetNoteValue();
+        // if mapWallet doesn't contain the spent outpoint's txid,
+        // the wallet is corrupt, so using `.at` here is fine.
+
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("type", ADDR_TYPE_ORCHARD);
+        entry.pushKV("spend", (int) actionIdx);
+        entry.pushKV("txidPrev", outpoint.hash.GetHex());
+        entry.pushKV("outputPrev", (int) outpoint.n);
+        auto ua = pwalletMain->FindUnifiedAddressByReceiver(receivedAt);
+        assert(ua.has_value());
```

Comment:
I assume this is because we can't import UFVKs at present? Add a TODO referencing the issue about adding UFVK imports.